File: Concurrency/Promises.py
import asyncio


async def fetch_data(url):
    await asyncio.sleep(1)
    if url.startswith("https"):
        return {"data": "Secure data from " + url}
    raise ValueError("Insecure URL")


async def main():
    try:
        result = await fetch_data("https://api.example.com")
        print(result["data"])
    except ValueError as err:
        print(str(err))
    finally:
        print("Done")


asyncio.run(main())

File: Concurrency/Promises.js
function fetchData(url) {
  return new Promise((resolve, reject) => {
    setTimeout(() => {
      if (url.startsWith("https")) {
        resolve({ data: "Secure data from " + url });
      } else {
        reject(new Error("Insecure URL"));
      }
    }, 1000);
  });
}

fetchData("https://api.example.com")
  .then(result => console.log(result.data))
  .catch(err  => console.error(err.message))
  .finally(()  => console.log("Done"));